core/src/main/java/com/redhat/lightblue/client/response/DefaultLightblueSingleValueResponse.java

```diff
@@ -0,0 +1,133 @@
+package com.redhat.lightblue.client.response;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.model.DataError;
+import com.redhat.lightblue.client.model.Error;
+import com.redhat.lightblue.client.util.JSON;
+
+/**
+ * <p>A special response type for responses from lightblue where only a single primitive value is expected.</p>
+ * <p><b>WARNING:</b> There should be only a few special cases where this response type is needed. It is not
+ * recommended for general usage.</p>
+ *
+ * @author dcrissman
+ */
+public class DefaultLightblueSingleValueResponse extends AbstractLightblueResponse implements LightblueResponse, LightblueErrorResponse {
```

Comment:
Yes, this is duplicate logic. I did this because DefaultLightblueSingleValueResponse is not a LightblueDataResponse. But you are right for the hand slap, let me see if there is a way I can abstract the logic so it is not duplicated.